src/StabilizerProcessor.cs

```diff
@@ -196,22 +196,29 @@ public override void OnDumpRegister<T>(T location, IQArray<Qubit> qubits)
 
         public override Result Measure(IQArray<Pauli> bases, IQArray<Qubit> qubits)
         {
-            if (!bases.TryGetSingleZ(out var idx))
+            if (qubits.Count == 1)
             {
-                // throw new UnsupportedOperationException("Not yet implemented.");
-                var aux = this.Simulator!.QubitManager?.Allocate();
-                if (aux == null) throw new NullReferenceException("Qubit manager was null.");
-                try
+                if (!bases.HasNonZ())
```

Comment:
I don't think that these checks do the same thing as `TryGetSingleZ`, basically its looking for the "easy" case of measurement where there is only one Z pauli and the rest are identity. Then you know that the identity measurements do nothing and you can MeasureByIndex the one that has a Z. Identities are fine, its just X and Y that will need to write to scratch.

Reply:
Hmmm.. MeasureOneTest gave that the original TryGetSingleZ, gave index 0 back, when asserting qubit index 1, which is the first Z. But because idx was 0, it measuring qubit 0... so... it needs some additional work then.